database/snowflake/snowflake.go

```diff
@@ -173,45 +180,72 @@ func (p *Snowflake) Unlock() error {
 }
 
 func (p *Snowflake) Run(migration io.Reader) error {
+
+	// run migration
+	if p.config.MultiStatementEnabled {
+		var err error
+		if e := multistmt.Parse(migration, multiStmtDelimiter, p.config.MultiStatementMaxSize, func(m []byte) bool {
+			tq := strings.TrimSpace(string(m))
+			if tq == "" {
+				return true
+			}
+			if _, err = p.conn.ExecContext(context.Background(), tq); err != nil {
```

Comment:
Refactor this to avoid duplicate code. Either move error handling into it's own method or query execution into it's own method.